app/controllers/users_controller.rb

```diff
@@ -1900,6 +1901,11 @@ def user_menu_messages
            }
   end
 
+  def lookup
+    users = User.where(username: params[:usernames])
+    render json: serialize_found_users(users)
+  end
```

Comment:
We need a limit here

Reply:
Did you think about this?

Reply:
I've limited it to 20 users max ans also tested it in 0233046df11ed7ff1be15098377338f9b8f678b1.